### PR TITLE
Fixed disabling of gradients in the torch code

### PR DESCRIPTION
**Pull request by haydn-jones**
`torch.no_grad()` is a context manager, it should be placed over a function as a decorator or used like so:
```python
with torch.no_grad():
    #  Computations below will be done without gradients
    x = torch.sigmoid(x)
```
Simply calling `no_grad()` will not disable gradients. The correct way to disable gradients globally would be 
```Python
torch.set_grad_enabled(False)
```

### CODE REVIEW

**Review by kuprel — APPROVED** [review]
Wow this significantly reduced the memory usage in torch, thanks!
